### ISSUE TITLE
[cmake] sphinx and read-only source directory (Trac #1056)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1050]
If sphinx is detected, it tries to write sphinx configuration in the source directory.  This fails if you don't have permission to write to the source.  Since docs are non-essential, this should just be a warning and disabling of docs.

Migrated from https://code.icecube.wisc.edu/ticket/1056
```json
{
    "status": "closed", 
    "changetime": "2019-01-11T23:55:13", 
    "description": "If sphinx is detected, it tries to write sphinx configuration in the source directory.  This fails if you don't have permission to write to the source.  Since docs are non-essential, this should just be a warning and disabling of docs.", 
    "reporter": "david.schultz", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1547250913615455", 
    "component": "cmake", 
    "summary": "[cmake] sphinx and read-only source directory", 
    "priority": "major", 
    "keywords": "sphinx, docs", 
    "time": "2015-07-16T19:10:54", 
    "milestone": "", 
    "owner": "david.schultz", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20150720T22:28:16: 
* david.schultz changed resolution from "" to "fixed"
* david.schultz changed owner from "" to "david.schultz"
* david.schultz changed status from "new" to "closed"
* david.schultz commented:

> In changeset 2171:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2171"
> fixes #1050
> ```

**Comment by icecube-trac** [comment]
Trac update at 20150722T15:53:26: 
* nega changed resolution from "fixed" to ""
* nega changed status from "closed" to "reopened"

**Comment by icecube-trac** [comment]
Trac update at 20150722T15:56:39: 
* nega commented:

> In changeset 2175:
> 
> ```text
> #!CommitTicketReference repository="IceTray" revision="2175"
> redo 2171/IceTray
> 
> fixes #1050
> ```


* nega changed status from "reopened" to "closed"
* nega changed resolution from "" to "fixed"

**Comment by icecube-trac** [comment]
Trac update at 20150810T20:31:17: 
* olivas commented:

> Milestone offline V15-07-00 deleted


* olivas changed milestone from "offline V15-07-00" to ""

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:54:48: nwhitehorn commented:

> In changeset 170055:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170055"
> fixes #1050
> ```

**Comment by icecube-trac** [comment]
Trac update at 20190111T23:55:13: nwhitehorn commented:

> In changeset 170059:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="170059"
> redo 2171/IceTray
> 
> fixes #1050
> ```
